pkg/eval/builtin_fn_misc_test.go

```diff
@@ -14,6 +14,19 @@ import (
 
 func TestKindOf(t *testing.T) {
 	Test(t,
+		That("kind-of x").Puts("string"),
+		That("kind-of []").Puts("list"),
+		That("kind-of [&]").Puts("map"),
+		That("kind-of $true").Puts("bool"),
+		That("kind-of (eq x y)").Puts("bool"),
+		That("kind-of {}").Puts("string"),
+		That("kind-of []{}").Puts("fn"),
+		That("kind-of (float64 1)").Puts("number"),
+		That("kind-of (num 1)").Puts("number"),
+		That("kind-of (num 1/3)").Puts("number"),
+		That("kind-of $buildinfo").Puts("structmap"),
+		That("kind-of (styled x bold)").Puts("ui:text"),
+		That("kind-of (styled-segment x)").Puts("ui:text-segment"),
```

Comment:
I agree but feel it warrants a comment. So I'm going to replace the above tests with this comment:

```
// This exists solely to verify the linkage between the `kind-of` Elvish command and the
// vals.Kind() function. We otherwise rely on the unit tests of that function.
```